ui/pages/confirmation/templates/add-ethereum-chain.js

```diff
@@ -331,7 +347,35 @@ function getValues(pendingApproval, t, actions, history) {
     ],
     cancelText: t('cancel'),
     submitText: t('approveButtonText'),
+    loadingText: t('addingCustomNetwork'),
     onSubmit: async () => {
+      let endpointChainId;
+      try {
+        endpointChainId = await jsonRpcRequest(customRpcUrl, 'eth_chainId');
+      } catch (_err) {
+        // eslint-disable-next-line no-console
+        console.error(
+          `Request for method 'eth_chainId on ${customRpcUrl} failed`,
+        );
+        return actions.rejectPendingApproval(
+          pendingApproval.id,
+          ethErrors.rpc.resourceUnavailable().serialize(),
+        );
+      }
+
+      if (pendingApproval.requestData.chainId !== endpointChainId) {
+        console.error(
+          `Chain ID returned by RPC URL ${customRpcUrl} does not match ${endpointChainId}`,
+        );
+        // actions.rejectPendingApproval(
+        //   pendingApproval.id,
+        //   ethErrors.rpc.invalidInput().serialize(),
+        // );
+        return {
+          error: [MISMATCHED_NETWORK_RPC_CHAIN_ID],
+        };
```

Comment:
This function now returns one of the following depending on if the function succeeds or fails:

* Above we return `actions.rejectPendingApproval` during a failure which is a [thunk object](https://github.com/MetaMask/metamask-extension/blob/33cc8d587aad05c0b41871ba3676676a3ce5680e/ui/store/actions.ts#L3731). (Note: I don't think this value is actually used anywhere)
* Here we return a javascript object containing the `errors` key with an array of strings
* Below we return a javascript object containing an empty`{}`

Is there a way we can consolidate these to provide a consistent return interface? Checking usage of the code, it seems the return value from this function hasn't used in the past, so we have some flexibility to set things up.